Add ESM support to Xerox CLI (#833)

phawxby opened the pull request:
@AndrewLeedham I presume this is the same as the XDS process, PR to release?
<!-- GITHUB_RELEASE PR BODY: prerelease-version -->
# Version

Published prerelease version: `@xerox/cli@0.2.0-next.1`
`@xerox/commitlint-config@2.3.1-next.0`
`@xerox/eslint-config@3.3.1-next.0`

<details>
  <summary>Changelog</summary>

  #### Feature
  
  - `@xerox/cli`
    - Add ESM support to Xerox CLI [#833](https://github.com/xeroxinteractive/config/pull/833) ([@phawxby](https://github.com/phawxby))
  
  #### Fix
  
  - chore: force release [#843](https://github.com/xeroxinteractive/config/pull/843) ([@phawxby](https://github.com/phawxby))
  - fix: revert normalize-url so esm conversion not required [#842](https://github.com/xeroxinteractive/config/pull/842) ([@phawxby](https://github.com/phawxby))
  
  #### Dependencies
  
  - chore(deps-dev): bump eslint from 7.28.0 to 7.31.0 [#840](https://github.com/xeroxinteractive/config/pull/840) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps): bump typescript from 4.3.2 to 4.3.5 [#839](https://github.com/xeroxinteractive/config/pull/839) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - `@xerox/eslint-config`
    - chore(deps): bump eslint-plugin-jsdoc from 35.1.3 to 36.0.6 [#838](https://github.com/xeroxinteractive/config/pull/838) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  
  #### Authors: 2
  
  - [@dependabot[bot]](https://github.com/dependabot[bot])
  - Paul Hawxby ([@phawxby](https://github.com/phawxby))
</details>
<!-- GITHUB_RELEASE PR BODY: prerelease-version -->

phawxby committed:
Add ESM support to Xerox CLI (#833)

* feat: add ESM support

* fix: vulnerability

AndrewLeedham commented:
> @AndrewLeedham I presume this is the same as the XDX process, PR to release?

Yeah, but it should release a next version, so you can test that all works first.

dependabot[bot] committed:
chore(deps): bump eslint-plugin-jsdoc from 35.1.3 to 36.0.6

Bumps [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc) from 35.1.3 to 36.0.6.
- [Release notes](https://github.com/gajus/eslint-plugin-jsdoc/releases)
- [Commits](https://github.com/gajus/eslint-plugin-jsdoc/compare/v35.1.3...v36.0.6)

---
updated-dependencies:
- dependency-name: eslint-plugin-jsdoc
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

AndrewLeedham committed:
Merge pull request #838 from xeroxinteractive/dependabot/npm_and_yarn/next/eslint-plugin-jsdoc-36.0.6

chore(deps): bump eslint-plugin-jsdoc from 35.1.3 to 36.0.6

dependabot[bot] committed:
chore(deps): bump typescript from 4.3.2 to 4.3.5

Bumps [typescript](https://github.com/Microsoft/TypeScript) from 4.3.2 to 4.3.5.
- [Release notes](https://github.com/Microsoft/TypeScript/releases)
- [Commits](https://github.com/Microsoft/TypeScript/compare/v4.3.2...v4.3.5)

---
updated-dependencies:
- dependency-name: typescript
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

AndrewLeedham committed:
Merge pull request #839 from xeroxinteractive/dependabot/npm_and_yarn/next/typescript-4.3.5

chore(deps): bump typescript from 4.3.2 to 4.3.5

dependabot[bot] committed:
chore(deps-dev): bump eslint from 7.28.0 to 7.31.0

Bumps [eslint](https://github.com/eslint/eslint) from 7.28.0 to 7.31.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.28.0...v7.31.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

AndrewLeedham committed:
Merge pull request #840 from xeroxinteractive/dependabot/npm_and_yarn/next/eslint-7.31.0

chore(deps-dev): bump eslint from 7.28.0 to 7.31.0

phawxby committed:
fix: revert normalize-url so esm conversion not required (#842)

* feat: add ESM support

* fix: vulnerability

* Revert "fix: vulnerability"

This reverts commit cec7280cce358ec9a21a1ed80f0b81298ee95721.

* fix: update normalize-url but don't convert to esm yet

* fix: actually V6 is fine. V7 is the problem

phawxby committed:
chore: force release (#843)

dependabot[bot] committed:
chore(deps): bump @typescript-eslint/eslint-plugin from 4.26.1 to 4.29.0

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 4.26.1 to 4.29.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.29.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

AndrewLeedham committed:
Merge pull request #844 from xeroxinteractive/dependabot/npm_and_yarn/next/typescript-eslint/eslint-plugin-4.29.0

chore(deps): bump @typescript-eslint/eslint-plugin from 4.26.1 to 4.29.0

dependabot[bot] committed:
chore(deps): bump @commitlint/config-conventional from 12.1.4 to 13.1.0

Bumps [@commitlint/config-conventional](https://github.com/conventional-changelog/commitlint) from 12.1.4 to 13.1.0.
- [Release notes](https://github.com/conventional-changelog/commitlint/releases)
- [Changelog](https://github.com/conventional-changelog/commitlint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/conventional-changelog/commitlint/compare/v12.1.4...v13.1.0)

---
updated-dependencies:
- dependency-name: "@commitlint/config-conventional"
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

AndrewLeedham committed:
Merge pull request #845 from xeroxinteractive/dependabot/npm_and_yarn/next/commitlint/config-conventional-13.1.0

chore(deps): bump @commitlint/config-conventional from 12.1.4 to 13.1.0

dependabot[bot] committed:
chore(deps): bump @typescript-eslint/parser from 4.26.1 to 4.29.0

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 4.26.1 to 4.29.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.29.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump tar from 4.4.13 to 4.4.15

Bumps [tar](https://github.com/npm/node-tar) from 4.4.13 to 4.4.15.
- [Release notes](https://github.com/npm/node-tar/releases)
- [Changelog](https://github.com/npm/node-tar/blob/main/CHANGELOG.md)
- [Commits](https://github.com/npm/node-tar/compare/v4.4.13...v4.4.15)

---
updated-dependencies:
- dependency-name: tar
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

AndrewLeedham committed:
Merge pull request #846 from xeroxinteractive/dependabot/npm_and_yarn/next/typescript-eslint/parser-4.29.0

chore(deps): bump @typescript-eslint/parser from 4.26.1 to 4.29.0

AndrewLeedham committed:
Merge pull request #847 from xeroxinteractive/dependabot/npm_and_yarn/tar-4.4.15

chore(deps): bump tar from 4.4.13 to 4.4.15

dependabot[bot] committed:
chore(deps): bump chalk from 4.1.1 to 4.1.2

Bumps [chalk](https://github.com/chalk/chalk) from 4.1.1 to 4.1.2.
- [Release notes](https://github.com/chalk/chalk/releases)
- [Commits](https://github.com/chalk/chalk/compare/v4.1.1...v4.1.2)

---
updated-dependencies:
- dependency-name: chalk
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

AndrewLeedham committed:
Merge pull request #848 from xeroxinteractive/dependabot/npm_and_yarn/next/chalk-4.1.2

chore(deps): bump chalk from 4.1.1 to 4.1.2

dependabot[bot] committed:
chore(deps): bump eslint-plugin-jest from 24.3.6 to 24.4.0

Bumps [eslint-plugin-jest](https://github.com/jest-community/eslint-plugin-jest) from 24.3.6 to 24.4.0.
- [Release notes](https://github.com/jest-community/eslint-plugin-jest/releases)
- [Changelog](https://github.com/jest-community/eslint-plugin-jest/blob/main/CHANGELOG.md)
- [Commits](https://github.com/jest-community/eslint-plugin-jest/compare/v24.3.6...v24.4.0)

---
updated-dependencies:
- dependency-name: eslint-plugin-jest
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

AndrewLeedham committed:
Merge pull request #849 from xeroxinteractive/dependabot/npm_and_yarn/next/eslint-plugin-jest-24.4.0

chore(deps): bump eslint-plugin-jest from 24.3.6 to 24.4.0

dependabot[bot] committed:
chore(deps): bump stylelint-scss from 3.19.0 to 3.20.1

Bumps [stylelint-scss](https://github.com/kristerkari/stylelint-scss) from 3.19.0 to 3.20.1.
- [Release notes](https://github.com/kristerkari/stylelint-scss/releases)
- [Changelog](https://github.com/kristerkari/stylelint-scss/blob/master/CHANGELOG.md)
- [Commits](https://github.com/kristerkari/stylelint-scss/compare/3.19.0...3.20.1)

---
updated-dependencies:
- dependency-name: stylelint-scss
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

AndrewLeedham committed:
Merge pull request #850 from xeroxinteractive/dependabot/npm_and_yarn/next/stylelint-scss-3.20.1

chore(deps): bump stylelint-scss from 3.19.0 to 3.20.1

dependabot[bot] committed:
chore(deps): bump postcss from 7.0.35 to 7.0.36

Bumps [postcss](https://github.com/postcss/postcss) from 7.0.35 to 7.0.36.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/7.0.35...7.0.36)

---
updated-dependencies:
- dependency-name: postcss
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump @commitlint/cli from 12.1.4 to 13.1.0

Bumps [@commitlint/cli](https://github.com/conventional-changelog/commitlint) from 12.1.4 to 13.1.0.
- [Release notes](https://github.com/conventional-changelog/commitlint/releases)
- [Changelog](https://github.com/conventional-changelog/commitlint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/conventional-changelog/commitlint/compare/v12.1.4...v13.1.0)

---
updated-dependencies:
- dependency-name: "@commitlint/cli"
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

AndrewLeedham committed:
Merge pull request #815 from xeroxinteractive/dependabot/npm_and_yarn/postcss-7.0.36

chore(deps): bump postcss from 7.0.35 to 7.0.36

AndrewLeedham committed:
Merge pull request #851 from xeroxinteractive/dependabot/npm_and_yarn/next/commitlint/cli-13.1.0

chore(deps): bump @commitlint/cli from 12.1.4 to 13.1.0

dependabot[bot] committed:
chore(deps-dev): bump prettier from 2.3.1 to 2.3.2

Bumps [prettier](https://github.com/prettier/prettier) from 2.3.1 to 2.3.2.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/2.3.1...2.3.2)

---
updated-dependencies:
- dependency-name: prettier
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

AndrewLeedham committed:
Merge pull request #852 from xeroxinteractive/dependabot/npm_and_yarn/next/prettier-2.3.2

chore(deps-dev): bump prettier from 2.3.1 to 2.3.2

dependabot[bot] committed:
chore(deps): bump eslint-plugin-testing-library from 4.6.0 to 4.10.1

Bumps [eslint-plugin-testing-library](https://github.com/testing-library/eslint-plugin-testing-library) from 4.6.0 to 4.10.1.
- [Release notes](https://github.com/testing-library/eslint-plugin-testing-library/releases)
- [Changelog](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/.releaserc.json)
- [Commits](https://github.com/testing-library/eslint-plugin-testing-library/compare/v4.6.0...v4.10.1)

---
updated-dependencies:
- dependency-name: eslint-plugin-testing-library
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

AndrewLeedham committed:
Merge pull request #853 from xeroxinteractive/dependabot/npm_and_yarn/next/eslint-plugin-testing-library-4.10.1

chore(deps): bump eslint-plugin-testing-library from 4.6.0 to 4.10.1

AndrewLeedham committed:
Resolve vulnerabilities

AndrewLeedham committed:
Upgrade to node 14

AndrewLeedham committed:
Merge pull request #955 from xeroxinteractive/AL-vuln

Resolve vulnerabilities

AndrewLeedham committed:
docs: remove dependabot from readmes

AndrewLeedham committed:
chore: remove dependabot

AndrewLeedham committed:
Merge pull request #956 from xeroxinteractive/AL-remove-dependabot

docs: remove dependabot from readmes

renovate-bot committed:
chore(deps): add renovate.json

AndrewLeedham committed:
Custom renovate config

AndrewLeedham committed:
Merge pull request #957 from xeroxinteractive/renovate/configure

Configure Renovate

renovate-bot committed:
Pin dependencies

AndrewLeedham committed:
Merge pull request #958 from xeroxinteractive/renovate/pin-dependencies

Pin dependencies

renovate-bot committed:
Pin dependency @types/react to v17.0.9

renovate-bot committed:
Update semantic-release monorepo

renovate-bot committed:
Update Node.js to v16

renovate-bot committed:
Update actions/setup-node action to v2

renovate-bot committed:
Update commitlint monorepo

AndrewLeedham committed:
Merge pull request #959 from xeroxinteractive/renovate/definitelytyped

Pin dependency @types/react to v17.0.9

AndrewLeedham committed:
Merge pull request #961 from xeroxinteractive/renovate/semantic-release-monorepo

Update semantic-release monorepo

renovate-bot committed:
Update all non-major dependencies

AndrewLeedham committed:
Merge pull request #966 from xeroxinteractive/renovate/major-commitlint-monorepo

Update commitlint monorepo (major)

AndrewLeedham committed:
Merge pull request #962 from xeroxinteractive/renovate/all-minor-patch

Update all non-major dependencies

AndrewLeedham committed:
Merge pull request #965 from xeroxinteractive/renovate/actions-setup-node-2.x

Update actions/setup-node action to v2

AndrewLeedham committed:
Merge pull request #964 from xeroxinteractive/renovate/node-16.x

Update Node.js to v16

renovate-bot committed:
Update dependency @types/react to v17.0.38

renovate-bot committed:
Update dependency eslint-plugin-jest to v25

AndrewLeedham committed:
Merge pull request #967 from xeroxinteractive/renovate/definitelytyped

Update dependency @types/react to v17.0.38

AndrewLeedham committed:
Merge pull request #968 from xeroxinteractive/renovate/eslint-plugin-jest-25.x

Update dependency eslint-plugin-jest to v25

renovate-bot committed:
Update typescript-eslint monorepo to v5

AndrewLeedham committed:
Merge pull request #969 from xeroxinteractive/renovate/major-typescript-eslint-monorepo

Update typescript-eslint monorepo to v5 (major)

renovate-bot committed:
Update dependency eslint-plugin-testing-library to v5

renovate-bot committed:
Update dependency eslint-plugin-jsdoc to v37

AndrewLeedham committed:
Merge pull request #972 from xeroxinteractive/renovate/eslint-plugin-jsdoc-37.x

Update dependency eslint-plugin-jsdoc to v37

renovate-bot committed:
Update dependency jest to v27

renovate-bot committed:
Update dependency eslint-plugin-prettier to v4

AndrewLeedham committed:
Merge pull request #971 from xeroxinteractive/renovate/eslint-plugin-prettier-4.x

Update dependency eslint-plugin-prettier to v4

AndrewLeedham committed:
Merge pull request #973 from xeroxinteractive/renovate/major-jest-monorepo

Update dependency jest to v27

AndrewLeedham committed:
Merge pull request #970 from xeroxinteractive/renovate/eslint-plugin-testing-library-5.x

Update dependency eslint-plugin-testing-library to v5

renovate-bot committed:
Update dependency stylelint-scss to v4

renovate-bot committed:
Update dependency stylelint-config-recommended to v6

renovate-bot committed:
Update semantic-release monorepo

AndrewLeedham committed:
Merge pull request #975 from xeroxinteractive/renovate/stylelint-scss-4.x

Update dependency stylelint-scss to v4

AndrewLeedham committed:
Merge pull request #977 from xeroxinteractive/renovate/stylelint-config-recommended-6.x

Update dependency stylelint-config-recommended to v6

renovate-bot committed:
Update dependency stylelint to v14

renovate-bot committed:
Update dependency stylelint-config-prettier to v9

renovate-bot committed:
Update dependency stylelint-prettier to v2

AndrewLeedham committed:
Merge pull request #974 from xeroxinteractive/renovate/stylelint-14.x

Update dependency stylelint to v14

AndrewLeedham committed:
Merge pull request #976 from xeroxinteractive/renovate/stylelint-prettier-2.x

Update dependency stylelint-prettier to v2

AndrewLeedham committed:
Merge pull request #978 from xeroxinteractive/renovate/major-semantic-release-monorepo

Update semantic-release monorepo (major)

AndrewLeedham committed:
Merge pull request #979 from xeroxinteractive/renovate/stylelint-config-prettier-9.x

Update dependency stylelint-config-prettier to v9

renovate-bot committed:
Update dependency normalize-url to v7

renovate-bot committed:
Update dependency semantic-release-slack-bot to v3

AndrewLeedham committed:
Merge pull request #981 from xeroxinteractive/renovate/normalize-url-7.x

Update dependency normalize-url to v7

renovate-bot committed:
Update dependency eslint to v8

AndrewLeedham committed:
Merge pull request #980 from xeroxinteractive/renovate/semantic-release-slack-bot-3.x

Update dependency semantic-release-slack-bot to v3

AndrewLeedham committed:
Resolve ESM in Jest + new ESLint testing

AndrewLeedham committed:
Merge pull request #982 from xeroxinteractive/renovate/eslint-8.x

Update dependency eslint to v8

renovate-bot committed:
Pin dependency enhanced-resolve to 5.8.3

AndrewLeedham committed:
Merge pull request #984 from xeroxinteractive/renovate/pin-dependencies

Pin dependency enhanced-resolve to 5.8.3

AndrewLeedham committed:
Add save-exact npmrc

AndrewLeedham committed:
Merge pull request #987 from xeroxinteractive/AL-npmrc

Add save-exact npmrc

AndrewLeedham committed:
Merge branch 'release' into next
